app/src/main/java/com/connfa/ui/fragment/LoadDataTask.java

```diff
@@ -0,0 +1,73 @@
+package com.connfa.ui.fragment;
+
+import android.os.AsyncTask;
+import android.support.annotation.Nullable;
+
+import com.connfa.model.Model;
+import com.connfa.model.managers.BofsManager;
+import com.connfa.model.managers.FavoriteManager;
+import com.connfa.model.managers.ProgramManager;
+import com.connfa.model.managers.SocialManager;
+import com.connfa.ui.drawer.DrawerManager;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+class LoadDataTask extends AsyncTask<Void, Void, List<Long>> {
+
+    private final DrawerManager.EventMode eventMode;
+    private final WeakReference<LoadDataTaskCallback> callbackRef;
```

Comment:
If this is held only by the activity you don't need a weak reference, mark & sweep removes isolated graphs